evm/src/fuzz/error.rs

```diff
@@ -0,0 +1,21 @@
+//! errors related to fuzz tests
+
+/// Magic return code for the `assume` cheatcode
+pub const ASSUME_MAGIC_RETURN_CODE: &[u8] = b"FOUNDRY::ASSUME";
+
+/// Possible errors when running fuzz tests
+#[derive(Debug, thiserror::Error)]
+pub enum FuzzError {
+    #[error("Couldn't call unknown contract")]
+    UnknownContract,
+    #[error("Couldn't find function")]
+    UnknownFunction,
+    #[error("Failed to decode fuzzer inputs")]
+    FailedDecodeInput,
+    #[error("Failed contract call")]
+    FailedContractCall,
+    #[error("Empty state changeset")]
+    EmptyChangeset,
+    #[error("The `vm.assume` cheatcode rejected too many inputs ({0} allowed)")]
+    TooManyRejects(u32),
+}
```

Comment:
are this used encoded inside evm?

because there's a `SolcError` trait that provides encoding functions

`impl SolError for FuzzError{}` is sufficient